cmd/snap-bootstrap/partition/encrypt.go

```diff
@@ -53,12 +59,30 @@ func (key EncryptionKey) Store(filename string) error {
 	// TODO:UC20: provision the TPM, generate and store the lockout authorization,
 	//            and seal the key. Currently we're just storing the unprocessed data.
 	if err := ioutil.WriteFile(filename, key[:], 0600); err != nil {
-		return fmt.Errorf("cannot store key file: %v", err)
+		return err
 	}
 
 	return nil
 }
 
+type RecoveryKey [recoveryKeySize]byte
+
+func NewRecoveryKey() (RecoveryKey, error) {
+	var key RecoveryKey
+	// rand.Read() is protected against short reads
+	_, err := rand.Read(key[:])
+	// On return, n == len(b) if and only if err == nil
+	return key, err
+}
+
+// Store writes the recovery key in the location specified by filename.
+func (key RecoveryKey) Store(filename string) error {
+	if err := os.MkdirAll(filepath.Dir(filename), 0755); err != nil {
```

Comment:
Does this need a TODO:UC20 comment like Store for the encryption key above?

Reply:
no